Sources/Utility/ArgumentParser.swift

```diff
@@ -565,9 +565,19 @@ public final class ArgumentParser {
             return results[argument.name] as? [T]
         }
 
-        /// Untyped argument by name
-        public func get(name: String) -> Any? {
-            return results[name]
+        /// Argument by name and type.
+        /// - Parameters:
+        ///   - name: the name of the argument.
+        ///   - type: The expected type of the parsed argument.
+        /// - throws: An ArgumentParserError error if the parsed argument does not match the type.
+        public func get<T>(_ name: String, type: T.Type) throws -> T? {
```

Comment:
There is no need to pass in the `T.Type`, Swift can infer this from the context.

For instances these two will compile removing ambiguity:

```swift
let verbose: Int = try args.get("--verbose")
let verbose = try args.get("--verbose") as Int
```

In most cases, the `as T` won't be necessary, because the context is good enough.

In your tests below, you will write:

```swift
XCTAssertEqual(try args.get("--verbose") as Int, 2)
XCTAssertEqual(try args.get("input files") as [String] ?? [], ["input1", "input2"])
XCTAssertEqual(try args.get("invalid") as Int, nil)
```

Reply:
I agree

Reply:
While I personally agree and prefer the approach without an explicit param as you guys do, I was trying to follow the style of the other generic API's that Apple has written as mentioned above, especially as it fits with the other ArgumentParser functions.
>Should the generic type be an explicit param though, like the Argument initializers (or Codable api's for that matter)?

Decodable
```
pathArgument = parser.add(option: "--path", kind: String.self)
```
OptionArgument
```
path = try values.decode(String.self, forKey: .path)
```
Both of these function calls could also infer the generic type through type inference, but don't

Reply:
Thoughts?

Reply:
The decision not to use inference for Codable was driven by the desire for better diagnostics in the face of a massive overload set and enforcing a single way to decode values (instead of let x = ... as T, let x : T = ..., let x = decode(T.type)).  We’re not exactly in either camp with this kind of API.

Reply:
Ok, good to know. How about the `parser.add` functions that have an explicit `kind` param?

Reply:
Also there to help with overload resolution.  The two `add` functions are potentially ambiguous without the metatype to distinguish between single and multiple arguments.

Reply:
Ok, we'll let @aciidb0mb3r decide. I'm happy either way! 👍

Reply:
I don't think it was possible to implement add method using type inference. However, the type can be inferred for `get` methods (at least right now), so lets go that way for now. It is also consistent with rest of the existing APIs.

Reply:
Changed to inferred type